src/core.ts

```diff
@@ -83,21 +83,21 @@ module.exports = {
       address: 'sync'
     }
   },
-  init: function (api, opts, permissions, manifest) {
+  init: function (api: any, opts: any, permissions: any, manifest: any) {
     // defaults
     //      opts.appKey = opts.appKey || appKey
 
-    var timeoutInactivity
+    var timeoutInactivity: number
     if (opts.timers && !isNaN(opts.timers.inactivity)) {
       timeoutInactivity = opts.timers.inactivity
     }
     // if opts.timers are set, pick a longer default
     // but if not, set a short default (as needed in the tests)
-    timeoutInactivity = timeoutInactivity || (opts.timers ? 600e3 : 5e3)
+    timeoutInactivity = timeoutInactivity! || (opts.timers ? 600e3 : 5e3)
```

Comment:
Why is this needed?

Reply:
Oh, `!` at the end of an identifier are a TypeScript-only feature. It may look weird since it reminds the `!` "not" operator.

`timeoutInactivity` was declared a `number` but it may not be assigned (if the `if` doesn't trigger), so TypeScript was warning that I can't "use" the value of `timeoutInactivity` yet. `timeoutInactivity!` tells TypeScript to ignore the risks of it not being assigned yet, and just "trust me". It turns out to compile to just `timeoutInactivity`.

There's an alternative solution to this which is declaring `timeoutInactivity` as `number | undefined`. But I didn't want that because code below the lines 96 will always see a `number`, never `undefined`, so I didn't want the rest of the code to handle undefined if it's never undefined. So essentially lines 90–96 comprise the declaration **and assignment** of `timeoutInactivity` and we are sure that after line 96 it's always a `number`.

Reply:
Alright, thanks for the explanation. Makes sense.